### ISSUE TITLE
Standardize PriorPose2 to also take Distibution objects during construction

### ISSUE THREAD

**Issue by dehann** [JuliaRobotics/RoME.jl#72]
Should be, for example:
```julia
PriorPose2(Distributions.MvNormal(zeros(3), 0.01*eye(3)))
```

See recent similar changes to `Pose2Pose2`: 
```julia
Pose2Pose2(Distributions.MvNormal(zeros(3), 0.01*eye(3)))
```
Will do this for `RoME v0.1.5` and influences the Hexagonal 2D SLAM examples.

cc @keevindoherty, @GearsAD

**Comment by dehann** [comment]
Will also change `PackedPriorPose2`

**Comment by dehann** [comment]
Must remove `compare` function duplicate from `RoME` after #77 is merged and `IncrementalInference v0.4.0`is tagged with change include JuliaRobotics/IncrementalInference.jl#113.

**Comment by dehann** [comment]
see documentation update here: JuliaRobotics/Caesar.jl#134
